### PR TITLE
Draft-example3

### PR DESCRIPTION
**Pull request by HesamShaelaie**
Could you please see this and give me your feedback.

### CODE REVIEW

File: docs/src/examples/FOBP_example2.jl [hunk 1]
```diff
@@ -1,90 +0,0 @@
-# # Foundations of Bilevel Programming: Example 2
```
Comment (odow) [inline docs/src/examples/FOBP_example2.jl:1]: You probably didn't mean to delete this file?

File: docs/src/examples/FOBP_example3.jl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+using BilevelJuMP
+using Ipopt
+using JuMP
+using Test
+
+model = BilevelModel(Ipopt.Optimizer, mode = BilevelJuMP.ProductMode(1e-9))
+
+# Global variables
+
+I = 7 # maximum literals
```
Comment (odow) [inline docs/src/examples/FOBP_example3.jl:44]: Now's good time to read the Julia performance tips:
https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-global-variables

**Review by odow — COMMENTED** [review]
@joaquimg 2 things:

1. CI is failing again. Not @HesamShaelaie fault.
2. Can you add him as a collaborator and turn on master branch protection? Then we can get doc previews: https://github.com/joaquimg/BilevelJuMP.jl/pull/129/checks?check_run_id=2898176202#step:5:560

File: docs/src/examples/FOBP_example2.jl (hunk 1 above)
Comment (HesamShaelaie) [reply docs/src/examples/FOBP_example2.jl:1]: sorry, I am learning.

File: docs/src/examples/FOBP_example3.jl (hunk 2 above)
Comment (HesamShaelaie) [reply docs/src/examples/FOBP_example3.jl:44]: Great, thanks. I assume you want to say don't use a global variable if you don't need to change in process and instead define constant variables.

**Comment by HesamShaelaie** [comment]
What should I do with the errors?

**Comment by odow** [comment]
@HesamShaelaie a trick: close and then reopen a pull request to restart the CI. @joaquim's PR #131  didn't hit the test failure so not sure what happened.

**Comment by joaquimg** [comment]
> @joaquimg 2 things:
> 
> 1. CI is failing again. Not @HesamShaelaie fault.
> 2. Can you add him as a collaborator and turn on master branch protection? Then we can get doc previews: https://github.com/joaquimg/BilevelJuMP.jl/pull/129/checks?check_run_id=2898176202#step:5:560

2. done
1. will check

**Comment by joaquimg** [comment]
All check passing now...

THe error was just Ipopt hitting:
`MathOptInterface.NUMERICAL_ERROR`
For no clear reason.
I have seen this one or twice.
Most time you can just re-run... maybe its a matter os improving initialization, not 100% sure....

File: docs/src/examples/FOBP_example2.jl (hunk 1 above)
Comment (odow) [reply docs/src/examples/FOBP_example2.jl:1]: You need to add it back and push a new commit. The easiest way is to checkout `master`, copy the file, then checkout `example3` and paste.

File: docs/src/examples/FOBP_example3.jl (hunk 2 above)
Comment (odow) [reply docs/src/examples/FOBP_example3.jl:44]: It's okay for this example.

**Review by odow — APPROVED** [review]

**Comment by odow** [comment]
Merging for now. We can fix any issues in a separate PR.

**Pull request merged**